File: BackEnd/AspNetWebApi/AspNetWebApi/Models/Produto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AspNetWebApi.Models
{
    public class Produto : BaseModelo
    {
        [Required(ErrorMessage = "Por favor, informe a Descrição."), MaxLength(250)]
        public string Descricao { get; set; }
        [Required(ErrorMessage = "Por favor, informe o Preço de Venda do produto.")]
        public decimal PrecoVenda { get; set; }

        [MaxLength]
        public string Imagem { get; set; }

        public int CodigoBarras { get; set; }

        [Required(ErrorMessage = "Por favor, informe a Categoria do produto.")]
        public ProdutoCategoria ProdutoCategoria { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }

    }
}

File: FrontEnd/Angular/src/app/classes/Produto.ts
import { Produtocategoria } from ".";

export class Produto {
  public Descricao: string;
  public PrecoVenda: number;
  public Imagem: string;
  public CodigoBarras: number;
  public ProdutoCategoriaId: number;
  public ProdutoCategoria?: Produtocategoria;
}
